src/graphnet/data/extractors/i3comparisonextractor.py

```diff
@@ -0,0 +1,30 @@
+"""I3Extractor class(es) for extracting SplineMPE reconstruction."""
+
+from typing import TYPE_CHECKING, Dict
+
+from graphnet.data.extractors.i3extractor import I3Extractor
+
+if TYPE_CHECKING:
+    from icecube import icetray  # pyright: reportMissingImports=false
+
+
+class I3ComparisonExtractor(I3Extractor):
+    """Class for extracting pointing predictions from various algorithms."""
+
+    def __init__(self, name: str):
+        """Construct I3ComparisonExtractor."""
+        # Base class constructor
+        super().__init__(name)
+
+    def __call__(self, frame: "icetray.I3Frame") -> Dict[str, float]:
+        """Extract pointing predictions."""
+        output = {}
+        if self._name in frame:
+            output.update(
+                {
+                    "zenith_" + self._name: frame[self._name].dir.zenith,
+                    "azimuth_" + self._name: frame[self._name].dir.azimuth,
```

Comment:
Consider adding other properties of I3Particles, e.g., energy, type, and position (might all be NaNs for most algorithms, but could still be useful to be able to extract these).